src/position.h

```diff
@@ -113,7 +113,7 @@ class Position {
   Bitboard attacks_from(PieceType pt, Square s) const;
   template<PieceType> Bitboard attacks_from(Square s) const;
   template<PieceType> Bitboard attacks_from(Square s, Color c) const;
-  Bitboard slider_blockers(Bitboard sliders, Square s, Bitboard& pinners) const;
+  Bitboard slider_blockers(Bitboard sliders, Square s, Bitboard* pinners = 0) const;
```

Comment:
Minor nit: you should use `nullptr` instead of 0 for null pointers in C++11.

Reply:
I know nullptr, that is why I did not use it as an initializer.
nullptr only makes sense when you call a function or instantiate a template. It helps avoid improper resolution. Here is an example with overloaded functions:
```
void f(void*) {
	printf("pointer\n");
}
void f(int) {
	printf("integer\n");
}
int main() {
	f(0);
	f(NULL);
	f(nullptr);
}
```
I am sorry for providing a primitive example if you know it already. But, based on the statement you made, I thought you might need it.

Reply:
Your argument for not using nullptr makes no sense whatsoever. Nullptr always makes sense for readability and correctness, it indicates to the reader of the source code that the programmer/author knows what he's doing and is assigning a null value to a pointer on purpose.

Reply:
@gvreuls summed it up nicely above - it is not strictly necessary here, but it is certainly a good practice to always use `nullptr` as null pointer constant. I'll add that GCC 4.7+ even has a warning for using 0 as null pointer value: `-Wzero-as-null-pointer-constant` and it triggers exactly on this kind of code (but it is not enabled by default even with `-Wall -Wextra`... I guess the world isn't quite ready for it yet :)

Reply:
> I guess the world isn't quite ready for it yet :)
True. The world is not yet 100% populated by dogmatic people like you.